### DISCUSSION TITLE
SAML-based access / retrieving a SAML assertion?

### DISCUSSION THREAD

**Discussion by freak4pc** [soto-project/soto#480] [Q&A]
Hey there, thank you for this great SDK. 
I'm trying to get a head start but having a hard time figuring out how to use SOTO. More specifically, how to obtain a SAML Assertion to be used with `STS.AssumeRoleWithSAMLRequest`, and what does `principalArn` represent. 

I remember you need to set your IdP somewhere to get the SAML assertion but I can't find the right API for this in SOTO.

Many thanks for your help!

P.S. Is there a Slack/Discord which would be better for these sort of questions?

**Comment by adam-fowler** [comment]
This is the link from the Soto-project front page for the Slack. https://join.slack.com/t/soto-project/shared_invite/zt-lo11z4ki-ebeGGykinDbjtWH1m~l6BA

Can't remember if it is up to date, if it isn't give us a shout.

**Reply by bok-** [reply]
@adam-fowler looks like this link is no longer valid. Is there a new invite somewhere?

**Reply by adam-fowler** [reply]
Have you tried the link on the soto-project home page? I just updated it

**Reply by bok-** [reply]
That worked, thanks!
